src/transport/HighPressureGasTransport.cpp

```diff
@@ -327,167 +323,579 @@ double HighPressureGasTransport::viscosity()
+    // Equation 9-5.23.
+    double ratio = MW_H/MW_L;
+    double A = 1.0;
     if (ratio > 9 && x_H > 0.05 && x_H < 0.7) {
-        FQ_mix_o *= 1 - 0.01*pow(ratio,0.87);
+        A = 1 - 0.01*pow(ratio,0.87);
     }
+    FQ_mix_o *= A;
 
-    // Calculate Z1m
-    double Z1m = (0.807*pow(Tr_mix,0.618) - 0.357*exp(-0.449*Tr_mix)
-                 + 0.340*exp(-4.058*Tr_mix)+0.018)*FP_mix_o*FQ_mix_o;
-
-    // Calculate Z2m:
-    double Z2m;
-    if (Tr_mix <= 1.0) {
-        if (Pr_mix < Pvp_mix/Pc_mix) {
-            double alpha = 3.262 + 14.98*pow(Pr_mix,5.508);
-            double beta = 1.390 + 5.746*Pr_mix;
-            Z2m = 0.600 + 0.760*pow(Pr_mix,alpha) + (0.6990*pow(Pr_mix,beta) -
-                0.60)*(1- Tr_mix);
-        } else {
-            throw CanteraError("HighPressureGasTransport::viscosity",
-                               "State is outside the limits of the Lucas model, Tr <= 1");
-        }
-    } else if ((Tr_mix > 1.0) && (Tr_mix < 40.0)) {
-        if ((Pr_mix > 0.0) && (Pr_mix <= 100.0)) {
-            double a_fac = 0.001245*exp(5.1726*pow(Tr_mix,-0.3286))/Tr_mix;
-            double b_fac = a_fac*(1.6553*Tr_mix - 1.2723);
-            double c_fac = 0.4489*exp(3.0578*pow(Tr_mix,-37.7332))/Tr_mix;
-            double d_fac = 1.7368*exp(2.2310*pow(Tr_mix,-7.6351))/Tr_mix;
-            double f_fac = 0.9425*exp(-0.1853*pow(Tr_mix,0.4489));
-
-            Z2m = Z1m*(1 + a_fac*pow(Pr_mix,1.3088)/(b_fac*pow(Pr_mix,f_fac)
-                        + pow(1+c_fac*pow(Pr_mix,d_fac),-1)));
-        } else {
-            throw CanteraError("HighPressureGasTransport::viscosity",
-                           "State is outside the limits of the Lucas model, 1.0 < Tr < 40");
-        }
-    } else {
-        throw CanteraError("HighPressureGasTransport::viscosity",
-                           "State is outside the limits of the Lucas model, Tr > 40");
-    }
 
-    // Calculate Y:
-    double Y = Z2m/Z1m;
+    params.FQ_mix_o = FQ_mix_o;
+    params.FP_mix_o = FP_mix_o;
+    params.Tc_mix = Tc_mix;
+    params.Tr_mix = Tr_mix;
+    params.Pc_mix = Pc_mix;
+    params.Pr_mix = Pr_mix;
+    params.MW_mix = MW_mix;
+    params.P_vap_mix = P_vap_mix;
 
-    // Return the viscosity:
-    return Z2m*(1 + (FP_mix_o - 1)*pow(Y,-3))*(1 + (FQ_mix_o - 1)
-            *(1/Y - 0.007*pow(log(Y),4)))/(ksi*FP_mix_o*FQ_mix_o);
 }
 
 // Pure species critical properties - Tc, Pc, Vc, Zc:
 double HighPressureGasTransport::Tcrit_i(size_t i)
 {
-    // Store current molefracs and set temp molefrac of species i to 1.0:
-    vector<double> molefracs = store(i, m_thermo->nSpecies());
+    vector<double> molefracs(m_thermo->nSpecies());
+    m_thermo->getMoleFractions(&molefracs[0]);
+    vector<double> mf_temp(m_thermo->nSpecies(), 0.0);
+    mf_temp[i] = 1;
+    m_thermo->setMoleFractions(&mf_temp[0]);
 
     double tc = m_thermo->critTemperature();
+
     // Restore actual molefracs:
     m_thermo->setMoleFractions(&molefracs[0]);
     return tc;
 }
```

Comment:
What would be the proper way to set these values during initialization? Have the constructor not be =default and set them in there?

Reply:
I followed the approach that the MixTransport class used with its "init" method instead of the constructor, and the issues I was having were resolved.